syft/frameworks/torch/crypto/securenn.py

```diff
@@ -43,26 +82,28 @@ def private_compare(x, r, BETA, j, alice, bob, crypto_provider):
         r (MultiPointerTensor): the threshold commonly held by alice and bob
         BETA (MultiPointerTensor): a boolean commonly held by alice and bob to
             hide the result of computation for the crypto provider
-        j (MultiPointerTensor): a tensor with 0 and 1 to to the loop in a
-            single pass
-        alice (AbstractWorker): 1st worker holding a private share
-        bob (AbstractWorker): 2nd worker holding a private share
-        crypto_provider (AbstractWorker): the crypto_provider
 
     return:
         β′ = β ⊕ (x > r).
     """
+    assert isinstance(x, sy.AdditiveSharingTensor)
+    assert isinstance(r, sy.MultiPointerTensor)
+    assert isinstance(BETA, sy.MultiPointerTensor)
+
+    alice, bob = x.locations
+    crypto_provider = x.crypto_provider
+    p = x.field
+    L = 2 ** Q_BITS  # 2**l
```

Comment:
Should this have 1 subtracted from it? L is not a prime field in this case

Reply:
It doesn't need to be prime (this value is set as is in the article)